lib/index.js

```diff
@@ -95,20 +95,33 @@ class Parser extends Transform {
       }
     }
     // Normalize option `delimiter`
+    const delimiter_error = new CsvError('CSV_INVALID_OPTION_DELIMITER', [
+      'Invalid option delimiter:',
+      'delimiter must be a non empty string or buffer or array of string|buffer,',
+      `got ${JSON.stringify(options.delimiter)}`
+    ])
     if(options.delimiter === undefined || options.delimiter === null || options.delimiter === false){
       options.delimiter = Buffer.from(',')
     }else if(typeof options.delimiter === 'string' && options.delimiter.length !== 0){
       options.delimiter = Buffer.from(options.delimiter)
+    }else if(Array.isArray(options.delimiter) && options.delimiter.length !== 0){
+      options.delimiter = options.delimiter.map(function(c){
+        if(c.length !== 0){
+          if(typeof c === 'string'){
+            return Buffer.from(c)
+          }else if(Buffer.isBuffer(c)){
+            return c;
+          }
+        }
+        throw delimiter_error
+      })
     }else if(
       (Buffer.isBuffer(options.delimiter) && options.delimiter.length === 0) ||
       (typeof options.delimiter === 'string' && options.delimiter.length === 0) ||
-      (!Buffer.isBuffer(options.delimiter) && typeof options.delimiter !== 'string')
+      (Array.isArray(options.delimiter) && options.delimiter.length === 0) ||
+      (!Buffer.isBuffer(options.delimiter) && typeof options.delimiter !== 'string' && !Array.isArray(options.delimiter))
     ){
-      throw new CsvError('CSV_INVALID_OPTION_DELIMITER', [
-        'Invalid option delimiter:',
-        'delimiter must be a non empty string or buffer,',
-        `got ${JSON.stringify(options.delimiter)}`
-      ])
+      throw delimiter_error
```

Comment:
There is root for simplication: 1st set a default value unless defined; 2nd convert it to an array, if not already one; then look through each element of the array and normalize/validate the element.

Reply:
ok so if I make sure that delimiter is always an array since the normalization then I can simply this code further? Is that right?